core/primitives/src/lib.rs

```diff
@@ -286,3 +286,25 @@ impl From<LogLevel> for log::Level {
 		}
 	}
 }
+
+/// Converts the given value into an `u64` by following the Substrate calling convention.
```

Comment:
I think a better wording would be "encodes the given value into a buffer and returns the pointer and length encoded as a single u64" or something like this.